src/shared/components/query/QueryStoreUtils.ts

```diff
@@ -52,13 +52,19 @@ export function queryParams(nonMolecularProfileParams:NonMolecularProfileQueryPa
 
 export function nonMolecularProfileParams(store:QueryStore):NonMolecularProfileQueryParams {
     const selectedStudyIds = store.allSelectedStudyIds;
+    let case_ids: string;
+    if (store.asyncCustomCaseSet.isComplete) {
+        case_ids = store.asyncCustomCaseSet.result.map(caseRow => (caseRow.studyId + ':' + caseRow.sampleId)).join('\r\n');
+    } else {
+        case_ids = store.caseIds;
+    }
```

Comment:
@adamabeshouse how does this fix the issue exactly? This was failing before because `asyncCustomCaseSet` wasn't complete ye?

Reply:
yes that's right, now instead we'll pass the caseIds as given from the page straight through